File: app/src/main/java/com/arthur/gloria/logired/features/trip/accepted/data/repository/AcceptedTripsRepositoryImpl.kt
package com.arthur.gloria.logired.features.trip.accepted.data.repository

import com.arthur.gloria.logired.core.network.ApiService
import com.arthur.gloria.logired.core.network.model.Trip
import com.arthur.gloria.logired.features.trip.accepted.domain.repository.AcceptedTripsRepository
import javax.inject.Inject

class AcceptedTripsRepositoryImpl @Inject constructor(
    private val api: ApiService
) : AcceptedTripsRepository {

    override suspend fun getAcceptedTrips(): Result<List<Trip>> {
        return try {
            val response = api.getMyAcceptedTrips()
            if (response.isSuccessful && response.body() != null) {
                Result.success(response.body()!!.rides ?: emptyList())
            } else {
                Result.failure(Exception("Error al obtener viajes aceptados: ${response.code()}"))
            }
        } catch (e: Exception) {
            Result.failure(Exception("Error de conexión: ${e.message}"))
        }
    }
}

File: lib/features/trip/accepted/data/repository/accepted_trips_repository_impl.dart
import '../../../../../core/di/service_locator.dart';
import '../../../../../core/network/model/models.dart';
import '../../domain/accepted_trips_repository.dart';

class AcceptedTripsRepositoryImpl implements AcceptedTripsRepository {
  @override
  Future<List<Trip>> getAcceptedTrips() async {
    final response = await sl.apiService.getMyAcceptedTrips();
    final list = response.data['rides'] as List? ?? [];
    return list.map((e) => Trip.fromJson(e)).toList();
  }
}
